docs/RFCS/range_leases.md

```diff
@@ -101,6 +114,23 @@ node ID and epoch match the last committed lease, the command will be
 applied; otherwise it will be rejected.
 
 
+# Performance implications
+
+We expect traffic proportional to the number of nodes in the system.
+With 1,000 nodes and a 9s liveness duration threshold, we expect every
+node to do a conditional put to update the heartbeat timestamp every
+7.2s. That would correspond to ~140 reqs/second, a not-unreasonable
+load for this function.
+
+We still require the traditional expiration-based range leases for any
+ranges located at or before the liveness table's range. This might be
+problematic in the case of meta2 address record ranges, which are
+expected to proliferate in a large cluster. This lease traffic could
```

Comment:
Would be nice to put a number to the number of ranges which couldn't use epoch-based range leases. How many meta2 ranges will there be for the 3.3 million range cluster example I mention? I'm not sure how big `RangeDescriptors` currently are (we should measure this), but let's assume 1KB per range of meta2 data. A single meta2 range could hold 64K which would translate into 50 meta2 ranges. That doesn't seem worth optimizing. 

Update: just checked on a local cluster and the size of a meta2 key+value averaged 184 bytes. That would translate into 364K ranges per meta2 range and we'd need 10 meta2 ranges to support 3.3 million ranges. You should probably double check my math here.